abci/types/types.proto

```diff
@@ -1,5 +1,6 @@
 syntax = "proto3";
-package types;
+package tendermint.types;
+option go_package = "github.com/tendermint/tendermint/abci/types";
```

Comment:
do you think we should also provide custom class names for other languages?

```
option csharp_namespace = "Tendermint.ABCI.Types";
option cc_enable_arenas = true;
option java_package = "com.tendermint.abci";
option java_outer_classname = "TypesProto";
option java_multiple_files = true;
option objc_class_prefix = "TMABCI";
option go_package = "github.com/tendermint/tendermint/abci/types;types";
```

similar to https://github.com/protocolbuffers/protobuf/blob/bb0c5439343a01ffbe9542f80f5297a7aedaba3b/src/google/protobuf/type.proto#L38

Reply:
also, don't `package tendermint.abci.types;` make more sense to you?

Reply:
For the namespace for other languages, I guess it's better, I'm not sure how it works in each case so I would rather keep the default but if if you have someone that manages the compatibility of your proto that would definitely be a good thing to do. 

I can prefix it with the package path. And maybe all of them, so something like that
- tendermint.abci.types
- tendermint.crypto.merkle
- tendermint.libs.common
- tendermint.rpc.grpc
- tendermint.types.proto3

We could have better names but at least this one is one decided by convention.

Let me know what you think.

Reply:
love the new names

Reply:
update done ;)